src/util/key_reader.py

```diff
@@ -0,0 +1,99 @@
+from config import LIST_ARDUINO_SERIAL_DEVICES_PATH
+from src.util.logger import get_logger
+import subprocess
+import serial
+import re
+
+logger = get_logger()
+
+class NoReaderFound(Exception):
```

Comment:
Should perhaps exchange the inherited error type to something more descriptive.

Reply:
Save it for a rainy day.

Reply:
Fixed